packages/react-query/src/useBaseQuery.ts

```diff
@@ -82,26 +82,27 @@ export function useBaseQuery<
       ),
   )
 
+  const [_, setForceUpdate] = React.useState(0)
+
   const result = observer.getOptimisticResult(defaultedOptions)
 
-  React.useSyncExternalStore(
-    React.useCallback(
-      (onStoreChange) => {
-        const unsubscribe = isRestoring
-          ? noop
-          : observer.subscribe(notifyManager.batchCalls(onStoreChange))
-
-        // Update result to make sure we did not miss any query updates
-        // between creating the observer and subscribing to it.
-        observer.updateResult()
-
-        return unsubscribe
-      },
-      [observer, isRestoring],
-    ),
-    () => observer.getCurrentResult(),
-    () => observer.getCurrentResult(),
-  )
+  React.useEffect(() => {
+    if (isRestoring) {
+      return
+    }
+
+    const unsubscribe = observer.subscribe(
+      notifyManager.batchCalls(() => {
+        setForceUpdate((n) => n + 1)
```

Comment:
should we go for a new object here?

```
setForceUpdate({})
```

pros:
- it's a bit shorter
- it won't create huge numbers

cons:
- creates a new object every time ??

Reply:
I don't see a downside with `setForceUpdate({})` - updated the code!